contracts/domainRegistry.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "./domainsSetLibrary.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "hardhat/console.sol";
+
+contract DomainRegistry is Ownable {
+    using domainSet for domainSet.Set;
+    using domainSet for domainSet.Domain;
+
+    receive() external payable {}
+
+    domainSet.Set internal domains;
+
+    event DomainRegistered(
+        string domainName,
+        address indexed owner,
+        uint256 deposit
+    );
+
+    constructor() Ownable(msg.sender) {}
+
+    function registerDomain(string memory domainName) external payable {
+        domainSet.Domain memory domain = domains.get(domainName);
+        require(domain.owner == address(0), "The domain already has an owner");
+        require(msg.value == domain.price, "Deposit must equal to collateral");
+        domains.update(
+            domainSet.Domain({
+                owner: msg.sender,
+                deposit: msg.value,
+                name: domainName,
+                price: domain.price
+            })
+        );
+        emit DomainRegistered(domainName, msg.sender, msg.value);
+    }
+
+    function withdrawAllFunds() external onlyOwner {
+        uint256 balance = address(this).balance;
+        require(balance > 0, "Contract has no balance to withdraw");
+        payable(owner()).transfer(balance);
+    }
+
+    function registerPriceDomain(
```

Comment:
What's the point of this function? You make the the domain controller address as zero address. Deposit as 0 as well. I've just got that it is for the reason not to return `deposit` to the msg.sender in the removeDomain. That does not make much sense as well.
You could achieve the same in a more simple way
-3 points